plz.el

```diff
@@ -359,13 +358,20 @@ NOQUERY is passed to `make-process', which see.
   (let* ((data-arg (pcase-exhaustive body-type
                      ('binary "--data-binary")
                      ('text "--data")))
+         (stream-options (pcase as
+                           (`(stream . ,options)
+                            (unless (plist-get options :through)
+                              (signal 'plz-error (make-plz-error :message "missing :through function")))
+                            (when (eq 'sync then)
```

Comment:
I added this to always go async, even if the `THEN` callback is not given:

```
(plz 'get "https://httpbin.org/stream/100"
  :as `(stream
        :through ,(lambda (process chunk)
                    (with-current-buffer (process-buffer process)
                      (message "%s" (point))
                      (message "Chunk:\n %s\n\n" (buffer-string))))))
```

Not setting this has the effect of streaming the response, calling the `PROCESS-FILTER`, and then waiting for the response. The response then looks like this:


```
#s(plz-response 2 200
                ((date . "Mon, 26 Feb 2024 09:00:49 GMT")
                 (content-type . "application/json")
                 (server . "gunicorn/19.9.0")
                 (access-control-allow-origin . "*")
                 (access-control-allow-credentials . "true"))
                nil #<process plz-request-curl>)
```

I think this could still be a useful mode to operate. Note that the body is nil in the case. The process buffer is also killed, so even if the `PROCESS-FILTER` would insert into the buffer, it would be lost.

Wdyt?

Reply:
Shouldn't the body always have the body?  I know it's not the case with your current implementation, but I'm not sure I understand why it shouldn't be populated.

Reply:
When streaming asynchronously I would say we should not accumulate. What if the response is very large, or keeps streaming for a very long time?